File: firstrun/devstral-glm/glm_4_7_flash/third_003_naive.py
import requests
import os
from typing import Dict, Any

def fetch_weather_data(city: str, api_key: str) -> Dict[str, Any]:
    """
    Fetches current weather data for a given city using the OpenWeatherMap API.

    Args:
        city (str): The name of the city to query.
        api_key (str): The API key for the weather service.

    Returns:
        Dict[str, Any]: A dictionary containing the weather data or an error message.
    """
    # Base URL for OpenWeatherMap Current Weather Data
    base_url = "https://api.openweathermap.org/data/2.5/weather"
    
    # Parameters for the API request
    params = {
        "q": city,
        "appid": api_key,
        "units": "metric"  # Use 'imperial' for Fahrenheit
    }

    try:
        response = requests.get(base_url, params=params)
        response.raise_for_status()  # Raises an HTTPError for bad responses (4xx or 5xx)
        return response.json()
    
    except requests.exceptions.HTTPError as http_err:
        return {"error": f"HTTP error occurred: {http_err}"}
    except requests.exceptions.ConnectionError as conn_err:
        return {"error": f"Connection error occurred: {conn_err}"}
    except requests.exceptions.Timeout as timeout_err:
        return {"error": f"The request timed out: {timeout_err}"}
    except requests.exceptions.RequestException as req_err:
        return {"error": f"An error occurred: {req_err}"}

File: latest_ran_output/qwen3_coder_next/third_003_naive.js
const axios = require('axios');

/**
 * Fetches current weather data for a given location using OpenWeatherMap API
 * @param {string} apiKey - Your OpenWeatherMap API key
 * @param {string} city - City name to get weather for
 * @returns {Promise<Object>} - Weather data object
 */
async function getWeatherData(apiKey, city) {
  try {
    const response = await axios.get('https://api.openweathermap.org/data/2.5/weather', {
      params: {
        q: city,
        appid: apiKey,
        units: 'metric' // Use metric units (Celsius)
      }
    });
    
    return {
      success: true,
      data: response.data
    };
  } catch (error) {
    if (error.response) {
      return {
        success: false,
        error: `API Error: ${error.response.status} - ${error.response.statusText}`,
        details: error.response.data
      };
    } else if (error.request) {
      return {
        success: false,
        error: 'No response received from API',
        details: error.request
      };
    } else {
      return {
        success: false,
        error: `Request setup error: ${error.message}`,
        details: error.config
      };
    }
  }
}

// Example usage (uncomment to test):
// getWeatherData('YOUR_API_KEY_HERE', 'London')
//   .then(result => console.log(result))
//   .catch(err => console.error(err));

module.exports = { getWeatherData };